torch/nn/functional.py

```diff
@@ -1607,6 +1608,8 @@ def nll_loss(input, target, weight=None, size_average=None, ignore_index=-100,
             out = torch._C._nn.nll_loss2d(
                 input, target, weight, reduction_enum, ignore_index)
             ret = out.view(out_size)
+    else:
```

Comment:
why is this necessary?

Reply:
In this case it's unnecessary because the `elif dim == 3 or dim > 4:` condition can be made into an else statement, since all other values of dim have already been handled. 

But in general we don't yet have any logic for exceptions in control-flow:
```
if cond:
    ret = 1
else:
    throw Exception()i
```

Since ret is not set in the else branch, the if node does not not emit it, even thought it should.